browser-test/src/admin/admin_program_migration.test.ts

```diff
@@ -58,42 +59,64 @@ test.describe('program migration', () => {
+      expect(
+        await adminProgramMigration.buttonEnabled(generateJSONButton),
+      ).toBe(false)
+      await validateScreenshot(page.locator('main'), 'export-page')
     })
 
-    await test.step('export program 2', async () => {
+    await test.step('generate json for program 2', async () => {
       await adminProgramMigration.selectProgramToExport('program-2')
-      const downloadedProgram = await adminProgramMigration.downloadProgram()
+      // Selecting a program enables the "Generate JSON" button
+      expect(
+        await adminProgramMigration.buttonEnabled(generateJSONButton),
+      ).toBe(true)
+
+      await adminProgramMigration.generateJSON()
+      const jsonPreview = await adminProgramMigration.expectJSONPreview()
+      expect(jsonPreview).toContain(programName)
+      expect(jsonPreview).toContain(block1Description)
+      expect(jsonPreview).toContain(block2Description)
+
+      // TODO(#7087): Once we export the questions, assert that all the questions
+      // are in the generated json.
+    })
+    await test.step('download json for program 2', async () => {
+      const downloadedProgram = await adminProgramMigration.downloadJSON()
       expect(downloadedProgram).toContain(programName)
       expect(downloadedProgram).toContain(block1Description)
       expect(downloadedProgram).toContain(block2Description)
       // TODO(#7087): Once we export the questions, assert that all the questions
       // are in the downloaded program.
     })
+
+    // TODO(#7582): Add a test to test that clicking the "Copy JSON" button works
   })
 
   test('import a program', async ({page, adminProgramMigration}) => {
     await test.step('load import page', async () => {
       await loginAsAdmin(page)
       await enableFeatureFlag(page, 'program_migration_enabled')
       await adminProgramMigration.goToImportPage()
-      await validateScreenshot(page, 'import-page-no-data')
+      await validateScreenshot(page.locator('main'), 'import-page-no-data')
     })
 
     await test.step('import a program', async () => {
-      const sampleJson = readFileSync(
+      const sampleJSON = readFileSync(
         'src/assets/import-program-sample.json',
         'utf8',
       )
-      await adminProgramMigration.submitProgramJson(sampleJson)
+      await adminProgramMigration.submitProgramJSON(sampleJSON)
 
       await adminProgramMigration.expectProgramImported('Import Sample Program')
       // The import page currently shows question IDs, so this screenshot needs
       // to be based on data that comes from a pre-created JSON file instead of
       // a runtime-downloaded JSON file, as the IDs could change at runtime.
       // Eventually, we likely won't show the question IDs and could take a
       // screenshot based on runtime-downloaded JSON.
-      await validateScreenshot(page, 'import-page-with-data')
+      await validateScreenshot(page.locator('main'), 'import-page-with-data')
```

Comment:
suggestion (non-blocking):  Can you try with the `fullPage` param set to `false`?  I think this might remove the header.  So, `await validateScreenshot(page.locator('main'), 'import-page-with-data', /* fullPage= */ false)`.

Reply:
@melanie-exygy I made this update, but it's still including the header 😞